js/style/style.js

```diff
@@ -453,6 +457,18 @@ Style.prototype = util.inherit(Evented, {
         }, () => {
             if (error) return callback(error);
 
+            var refFeatures = [];
+            if (Object.keys(this.refs).length) features.forEach((feature) => {
+                if (this.refs[feature.layer.id] && this.refs[feature.layer.id].length) {
+                    this.refs[feature.layer.id].forEach(refLayer => {
+                        var copiedFeature = JSON.parse(JSON.stringify(feature));
```

Comment:
You only need a one level deep clone right? `var copiedFeature = util.extend({}, feature)`.